game-core/src/main/java/games/strategy/engine/data/GameData.java

```diff
@@ -97,8 +95,10 @@
   private transient GameDataEventListeners gameDataEventListeners = new GameDataEventListeners();
 
   private void readObject(final ObjectInputStream in) throws IOException, ClassNotFoundException {
+    // The process of deserializing makes use of this lock,
+    // we'll get an NPE if we don't set this field here already.
+    readWriteLock = new ReentrantReadWriteLock();
```

Comment:
Serialization is stupid -_-
We can't use the final keyword anymore, but I think this is better than performing a null-check that's impossible to fail if it wasn't for (de)serialization